src/core/cuda_common_cuda.cu

```diff
@@ -221,8 +221,7 @@ void clear_energy_on_GPU() {
   cuda_safe_mem(cudaMemset(energy_device, 0, sizeof(CUDA_energy)));
 }
 
-void copy_energy_from_GPU() {
-  extern Observable_stat obs_energy;
+void copy_energy_from_GPU(Observable_stat &obs_energy) {
```

Comment:
I think it would be easier to return the gpu energy struct from the function, that way, the cuda interface code can be completely decoupled from the obssttat code, and avoid an output parameter.

Reply:
good idea